Base/src/main/java/io/deephaven/base/ringbuffer/CharRingBuffer.java

```diff
@@ -16,10 +16,21 @@
  * buffer instead.
  */
 public class CharRingBuffer implements Serializable {
+    private static int SYSTEM_ARRAYCOPY_THRESHOLD = 16;
     protected final boolean growable;
     protected char[] storage;
     protected int head, tail, size;
 
+    private void fastCopy(char[] src, int srcPos, char[] dest, int destPos, int length) {
+        if (length < SYSTEM_ARRAYCOPY_THRESHOLD) {
+            for (int ii = 0; ii < length; ii++) {
+                dest[destPos + ii] = src[srcPos + ii];
+            }
+        } else {
+            System.arraycopy(src, srcPos, dest, 0, length);
```

Comment:
You said you're reverting to array copy, but just in case:
```suggestion
            System.arraycopy(src, srcPos, dest, destPos, length);
```